--- a/ssl/s3_pkt.c
+++ b/ssl/s3_pkt.c
@@ -515,11 +515,6 @@ static int ssl3_get_record(SSL *s)
         }
 
         i = s->method->ssl3_enc->mac(s, md, 0 /* not send */ );
-        if (i < 0 || mac == NULL
-            || CRYPTO_memcmp(md, mac, (size_t)mac_size) != 0)
-            enc_err = -1;
-        if (rr->length > SSL3_RT_MAX_COMPRESSED_LENGTH + extra + mac_size)
-            enc_err = -1;
     }
 
     if (enc_err < 0) {
